docs/reference/concepts/hyperscaling.md

```diff
@@ -1,152 +1,246 @@
 
-- Sequential transaction processing will always be a bottleneck. At least some transactions will need to interact with the same storage slots of the shared state, and no amount of parallelism can help.
-- Decentralized p2p networks and consensus mechanisms will have their own bottlenecks too.
-- Data availability will have a bottleneck, at least until advanced sharding is implemented at L1 level.
-- Users have different preferences on conflicting requirements: degree of decentralization vs. latency, privacy vs. transparency, generic composability vs. ultra-high security vs. ultra-low costs, and so on. No particular choice can be optimal for all use-cases.
+The current state of bridging solutions is a [mess](https://twitter.com/VitalikButerin/status/1479501366192132099). There are some good solutions, such as atomic swaps, which enable trustless asset swapping between different blockchains. Unfortunately, they are not suitable for general message passing.
 
-Luckily, ZKPs offer a beautiful way to build a heterogeneous yet simultaneously a hyper-scalable blockchain system. This idea is known as [Fractal scaling](https://medium.com/starkware/fractal-scaling-from-l2-to-l3-7fe238ecfb4f).
-Many different ZKP chains (in the zkSync world we call them Hyperchains) are run in parallel and get their block proofs aggregated into a single final block that will be settled on L1. Each of the Hyperchains will resemble the entire system (i.e. it can have an infinite number of other Hyperchains on top of it: L3, L4, and so on).
+<div  align="center">
 
-![Hyperchains!](../../assets/images/hyperscaling5.png "Hyperchains")
+![](../../assets/images/hyperscalingIslands.png)
 
-Fractal scaling is necessary but not sufficient to achieve hyperscaling. You need one additional component:
+</div>
 
-:::tip What are Hyper-bridges?
+In order to pass information between chains, bridges based on economic incentives were introduced. These require the operators to lock funds, which can be slashed based on an arbitration process.
 
-Hyper-bridges = native bridges that enable transfers between any two Hyperchains without consuming resources on a third one.
+The potential design space is vast, and there are several potential issues. These include:
 
-:::
+1. The arbitration process is not trustless; dapp developers have to consider the risks involved.
+2. These bridges usually do not transfer real assets cross-chain; they only free up already locked funds. This makes the process expensive. However, this can be solved by using general message passing and minting synthetic assets. But the synthetic assets are less secure than the original assets and are not interchangeable with them.
+3. Chains can hard fork. This is their right, but it endangers the entire bridging ecosystem. This can be mitigated by locking funds separately for each bridge, but this makes the process much more expensive due to liquidity fragmentatio This applies to ZK-enabled bridges as well.
```

Comment:
```suggestion
3. Chains can hard fork. This is their right, but it endangers the entire bridging ecosystem. This can be mitigated by locking funds separately for each bridge, but this makes the process much more expensive due to liquidity fragmentation. This applies to ZK-enabled bridges as well.
```